script.js

```diff
@@ -1,55 +1,68 @@
-function Animal(name){
+class Animal {
+  constructor(name) {
     this.name = name;
-}
-
-Animal.prototype.action = function(){
-  document.getElementById(this.actionSoundName).play();
-  
-}
+  }
 
-Animal.prototype.putInTheDocument = function(){
+  action() {
+    document.getElementById(this.actionSoundName).play();
+  }
+  addimage() {
+    var t = document.querySelector("body").lastElementChild;
+    if (t == document.querySelector("body").querySelector("img")) t.remove();
+    var petImg = document.createElement("img");
+    petImg.src = this.imgurl;
+    document.querySelector("body").appendChild(petImg);
+  }
+  putInTheDocument() {
     var petsTable = document.getElementById("petsTable");
     var petTR = document.createElement("tr");
-  
     var petNameTD = document.createElement("td");
+    petNameTD.classList.add("hov");
     petNameTD.textContent = this.name;
     petTR.appendChild(petNameTD);
-  
+
     var petLegsTD = document.createElement("td");
+    petLegsTD.classList.add("hov");
+    //petLegsTD.setAttribute("id","demoid");
+
     petLegsTD.textContent = this.legs;
     petTR.appendChild(petLegsTD);
-  
+
     var petActionTD = document.createElement("td");
     var petActionTDButton = document.createElement("button");
     petActionTDButton.textContent = this.actionText;
     petActionTD.appendChild(petActionTDButton);
     petTR.appendChild(petActionTD);
-    
+
     petActionTDButton.onclick = this.action.bind(this);
-    petsTable.querySelector("tbody").appendChild(petTR)
+    petsTable.querySelector("tbody").appendChild(petTR);
+    petNameTD.onclick = this.addimage.bind(this);
```

Comment:
Birden fazla td olabilirdi. Bu durumda tek tek onclick yazman gerekecekti. Tr'ye onclick atayip Buttonlarin calistirildigi methotta(action() methodu) stopPropagation kullanbilirsin.

Reply:
Öneriniz için çok teşekkür ederim.Muhakkak ki haklsınız,dikkate alacağım.Ödevim hakkında ki düşünceleriniz için ayrıca müteşekkirim. :)